packages/ensnode-react/src/hooks/useIndexingStatus.ts

```diff
@@ -18,12 +26,42 @@ export function useIndexingStatus(
 
   const queryOptions = createIndexingStatusQueryOptions(_config);
 
+  const cachedSnapshotRef = useRef<CrossChainIndexingStatusSnapshotOmnichain | null>(null);
+
   const options = {
     ...queryOptions,
     refetchInterval: 10 * 1000, // 10 seconds - indexing status changes frequently
     ...query,
     enabled: query.enabled ?? queryOptions.enabled,
   };
 
-  return useQuery(options);
+  const queryResult = useQuery(options);
+
+  // Update cached snapshot whenever we get a successful response
```

Comment:
```suggestion
  // Store the latest snapshot in memory whenever we get a successful response.
  // Each incremental snapshot will have >= indexing progress than previous snapshots.
```

Based on Slack chat with @lightwalker-eth this is probably better comment